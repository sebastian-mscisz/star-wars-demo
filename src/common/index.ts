export * from './endpoints';
export * from './paths';
export * from './types';
export * from './utils';
