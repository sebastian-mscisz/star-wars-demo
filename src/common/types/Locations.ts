import { DataItemWithNeighbours } from './Common';

export type Location = {
    name?: string;
    url?: string;
    residents?: string[];
    population?: string;
};

export type LocationWithNeighbours = DataItemWithNeighbours<Location>;

export type Locations = Location[];
