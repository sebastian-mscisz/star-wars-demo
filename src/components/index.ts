export * from './character-details';
export * from './characters';
export * from './characters-card';
export * from './error-component';
export * from './items-with-pagination';
export * from './location-details';
export * from './locations';
export * from './locations-card';
export * from './marginal-components';
export * from './pagination-buttons';
export * from './spinner';
export * from './vehicle-details';
export * from './vehicles';
export * from './vehicles-card';
