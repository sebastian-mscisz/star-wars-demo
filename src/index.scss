body {
    min-height: 100vh;
    display: flex;
    flex-direction: column;
    padding: 0;
    margin: 0;
    background-image: url('/wallpaper.jpg');
    background-repeat: no-repeat;
    background-color: black;
    width: 100%;
    font-family: sans-serif;
}

#root {
    display: flex;
    flex-wrap: wrap;
    flex-grow: 1;
}
