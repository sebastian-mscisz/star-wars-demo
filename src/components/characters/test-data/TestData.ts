import { Characters, Character, Species } from 'src/common';

export const TestCharacter: Character = {
    name: 'Character1',
    url: 'https://swapi.info/api/people/1',
    homeworld: 'https://swapi.info/api/planets/1',
    species: ['species/1'],
    vehicles: ['https://swapi.info/api/vehicles/1'],
};

export const TestCharacters: Characters = [
    TestCharacter,
    {
        name: 'Character2',
        url: 'https://swapi.info/api/people/2',
        vehicles: ['https://swapi.info/api/vehicles/1', 'https://swapi.info/api/vehicles/2'],
        homeworld: 'https://swapi.info/api/planets/2',
        species: ['species/2'],
    },
    {
        name: 'Character3',
        url: 'https://swapi.info/api/people/3',
        vehicles: [],
        homeworld: 'https://swapi.info/api/planets/3',
        species: ['species/3'],
    },
    {
        name: 'Character4',
        url: 'https://swapi.info/api/people/4',
        vehicles: ['https://swapi.info/api/vehicles/3'],
        homeworld: 'https://swapi.info/api/planets/4',
        species: ['species/4'],
    },
    {
        name: 'Character5',
        url: 'https://swapi.info/api/people/5',
        vehicles: ['https://swapi.info/api/vehicles/4', 'https://swapi.info/api/vehicles/5'],
        homeworld: 'https://swapi.info/api/planets/4',
        species: ['species/5'],
    },
];

export const TestSpecies: Species = [
    {
        name: 'Species2',
        url: 'species/2',
    },
    {
        name: 'Species2',
        url: 'species/2',
    },
    {
        name: 'Species3',
        url: 'species/3',
    },
    {
        name: 'Species4',
        url: 'species/4',
    },
    {
        name: 'Species5',
        url: 'species/5',
    },
];
