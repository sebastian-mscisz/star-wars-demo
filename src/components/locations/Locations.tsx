import { withHandlers } from 'src/hoc';
import { LocationsUi } from './ui/LocationsUi';
import { useLocations } from 'src/hooks';

const LocationsWithHandlers = withHandlers(LocationsUi);

export const Locations: React.FC = () => {
    const { data: locations, isLoading, loadingError } = useLocations();

    return <LocationsWithHandlers isLoading={isLoading} locations={locations} hasError={!!loadingError} />;
};
