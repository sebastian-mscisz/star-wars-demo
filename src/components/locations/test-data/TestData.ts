import { Locations, Location } from 'src/common';

export const TestLocation: Location = {
    name: 'Location1',
    url: 'https://swapi.info/api/planets/1',
    residents: ['https://swapi.info/api/people/1'],
    population: '1000',
};

export const TestLocations: Locations = [
    TestLocation,
    {
        name: 'Location2',
        url: 'https://swapi.info/api/planets/2',
        residents: ['https://swapi.info/api/people/2'],
        population: '5343',
    },
    {
        name: 'Location3',
        url: 'https://swapi.info/api/planets/3',
        residents: ['https://swapi.info/api/people/3'],
        population: '13333',
    },
    {
        name: 'Location4',
        url: 'https://swapi.info/api/planets/4',
        residents: ['https://swapi.info/api/people/4', 'https://swapi.info/api/people/5'],
        population: '9999',
    },
    {
        name: 'Location5',
        url: 'https://swapi.info/api/planets/5',
        residents: [],
        population: '999',
    },
];
