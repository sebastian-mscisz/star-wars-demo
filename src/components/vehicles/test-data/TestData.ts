import { Vehicles, Vehicle } from 'src/common';

export const TestVehicle: Vehicle = {
    name: 'Vehicle1',
    url: 'https://swapi.info/api/vehicles/1',
    vehicle_class: 'Air speeder',
    pilots: ['character/1', 'character/2'],
};

export const TestVehicles: Vehicles = [
    TestVehicle,
    {
        name: 'Vehicle2',
        url: 'https://swapi.info/api/vehicles/2',
        pilots: ['character/2'],
        vehicle_class: 'Land speeder',
    },
    {
        name: 'Vehicle3',
        url: 'https://swapi.info/api/vehicles/3',
        pilots: ['character/4'],
        vehicle_class: 'Water speeder',
    },
    {
        name: 'Vehicle4',
        url: 'https://swapi.info/api/vehicles/4',
        pilots: ['character/5'],
        vehicle_class: 'Fire speeder',
    },
    {
        name: 'Vehicle5',
        url: 'https://swapi.info/api/vehicles/5',
        pilots: ['character/5'],
        vehicle_class: 'Air speeder',
    },
];
