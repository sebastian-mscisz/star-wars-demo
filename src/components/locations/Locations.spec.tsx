import { BrowserRouter } from 'react-router-dom';
import { Locations } from './Locations';
import { screen, render } from '@testing-library/react';
import MockAdapter from 'axios-mock-adapter';
import { DataApi } from 'src/xhr';
import { getLocationsEndpoint } from 'src/common';
import { TestLocations } from './test-data';
import { Provider } from 'react-redux';
import { createStore } from '../../store';

describe('Locations', () => {
    let dataApiMock: MockAdapter;

    beforeEach(() => {
        dataApiMock = new MockAdapter(DataApi);
    });

    it('should render', async () => {
        dataApiMock.onGet(getLocationsEndpoint()).reply(200, TestLocations);

        const { asFragment } = render(
            <Provider store={createStore()}>
                <Locations />
            </Provider>,
            { wrapper: BrowserRouter },
        );

        await screen.findByText('Location2');

        expect(asFragment()).toMatchSnapshot();
    });
});
