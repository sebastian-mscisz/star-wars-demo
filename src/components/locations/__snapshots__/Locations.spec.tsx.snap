// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Locations should render 1`] = `
<DocumentFragment>
  <div
    class="sw-pagination-section"
    data-testid="pagination-section"
  >
    <div
      class="sw-pagination-button sw-pagination-button--hidden"
      data-testid="left-pagination-button"
    >
      ←
    </div>
    <div
      class="sw-pagination-button sw-pagination-button--hidden"
      data-testid="right-pagination-button"
    >
      →
    </div>
  </div>
  <div
    class="sw-locations"
  >
    <div
      class="sw-locations-card"
    >
      <div
        class="sw-locations-card__tile"
      >
        <a
          class="sw-locations-card__tile__link"
          href="/location/1"
        >
          <h5
            class="sw-locations-card__tile__link__text"
          >
            Location1
          </h5>
          <div
            class="sw-locations-card__tile__link__image-wrapper"
          >
            <img
              alt="Star wars planet"
              src="/locations/1.jpg"
            />
          </div>
        </a>
      </div>
    </div>
    <div
      class="sw-locations-card"
    >
      <div
        class="sw-locations-card__tile"
      >
        <a
          class="sw-locations-card__tile__link"
          href="/location/2"
        >
          <h5
            class="sw-locations-card__tile__link__text"
          >
            Location2
          </h5>
          <div
            class="sw-locations-card__tile__link__image-wrapper"
          >
            <img
              alt="Star wars planet"
              src="/locations/2.jpg"
            />
          </div>
        </a>
      </div>
    </div>
    <div
      class="sw-locations-card"
    >
      <div
        class="sw-locations-card__tile"
      >
        <a
          class="sw-locations-card__tile__link"
          href="/location/3"
        >
          <h5
            class="sw-locations-card__tile__link__text"
          >
            Location3
          </h5>
          <div
            class="sw-locations-card__tile__link__image-wrapper"
          >
            <img
              alt="Star wars planet"
              src="/locations/3.jpg"
            />
          </div>
        </a>
      </div>
    </div>
    <div
      class="sw-locations-card"
    >
      <div
        class="sw-locations-card__tile"
      >
        <a
          class="sw-locations-card__tile__link"
          href="/location/4"
        >
          <h5
            class="sw-locations-card__tile__link__text"
          >
            Location4
          </h5>
          <div
            class="sw-locations-card__tile__link__image-wrapper"
          >
            <img
              alt="Star wars planet"
              src="/locations/4.jpg"
            />
          </div>
        </a>
      </div>
    </div>
    <div
      class="sw-locations-card"
    >
      <div
        class="sw-locations-card__tile"
      >
        <a
          class="sw-locations-card__tile__link"
          href="/location/5"
        >
          <h5
            class="sw-locations-card__tile__link__text"
          >
            Location5
          </h5>
          <div
            class="sw-locations-card__tile__link__image-wrapper"
          >
            <img
              alt="Star wars planet"
              src="/locations/5.jpg"
            />
          </div>
        </a>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
